tools/smoother_benchmarking/README.md

```diff
@@ -0,0 +1,121 @@
+# Planners Smoothing Benchmark
+
+This experiment runs a set with randomly generated goals for objective benchmarking.
+
+Bechmarking scripts require the following python packages to be installed:
+
+```
+pip install transforms3d
+pip install seaborn
+pip install tabulate
+```
+
+To perform the benchmarking, the following changes should be made for planner and smoother servers:
+
+```
+diff --git a/nav2_planner/src/planner_server.cpp b/nav2_planner/src/planner_server.cpp
+index c7a90bcb..510bb6ea 100644
+--- a/nav2_planner/src/planner_server.cpp
++++ b/nav2_planner/src/planner_server.cpp
+@@ -285,7 +285,7 @@ bool PlannerServer::getStartPose(
+   typename std::shared_ptr<const typename T::Goal> goal,
+   geometry_msgs::msg::PoseStamped & start)
+ {
+-  if (goal->use_start) {
++  if (true) {
+     start = goal->start;
+   } else if (!costmap_ros_->getRobotPose(start)) {
+     return false;
+@@ -381,7 +381,10 @@ void PlannerServer::computePlanThroughPoses()
+       }
+ 
+       // Get plan from start -> goal
++      auto planning_start = steady_clock_.now();
+       nav_msgs::msg::Path curr_path = getPlan(curr_start, curr_goal, goal->planner_id);
++      auto planning_duration = steady_clock_.now() - planning_start;
++      result->planning_time = planning_duration;
+ 
+       if (!validatePath<ActionThroughPoses>(curr_goal, curr_path, goal->planner_id)) {
+         throw nav2_core::NoValidPathCouldBeFound(goal->planner_id + "generated a empty path");
+@@ -398,7 +401,7 @@ void PlannerServer::computePlanThroughPoses()
+     publishPlan(result->path);
+ 
+     auto cycle_duration = steady_clock_.now() - start_time;
+-    result->planning_time = cycle_duration;
++    // result->planning_time = cycle_duration;
+ 
+     if (max_planner_duration_ && cycle_duration.seconds() > max_planner_duration_) {
+       RCLCPP_WARN(
+diff --git a/nav2_smoother/src/nav2_smoother.cpp b/nav2_smoother/src/nav2_smoother.cpp
+index ada1f664..610e9512 100644
+--- a/nav2_smoother/src/nav2_smoother.cpp
++++ b/nav2_smoother/src/nav2_smoother.cpp
+@@ -253,8 +253,6 @@ bool SmootherServer::findSmootherId(
+ 
+ void SmootherServer::smoothPlan()
+ {
+-  auto start_time = steady_clock_.now();
```

Comment:
Why move this?

Reply:
This was moved for the same [reasons](https://github.com/ros-planning/navigation2/pull/3236#discussion_r992016301).